spec/features/user_edit_spec.rb

```diff
@@ -1,61 +1,77 @@
 require 'rails_helper'
 
-describe 'edit users' do
-  let(:base_attributes) { { first_name: 'John', last_name: 'Smith' } }
+describe 'edit staff users' do
+  let!(:user) { create :user, staff: true, last_name: 'Smith' }
 
-  before :each do
-    when_current_user_is :staff, integration: true
-    @user = create :user, base_attributes
-    visit edit_user_path(@user)
+  context 'clicking from dashboard' do
+    before :each do
+      when_current_user_is :staff, integration: true
+      visit staff_dashboard_url
+    end
+    it 'directs to the correct page' do
+      click_link user.proper_name.to_s, href: edit_user_path(user)
+      expect(page.current_url).to eql edit_user_url(user)
+    end
   end
 
-  context 'every field is filled in correctly' do
+  context 'on edit user page' do
     before :each do
-      within 'form.edit_user' do
-        fill_in_fields_for User, attributes: base_attributes
-          .merge(first_name: 'Bananas')
-      end
+      when_current_user_is :staff, integration: true
+      visit edit_user_url(user)
     end
 
-    it 'only changes the given attributes' do
-      click_on 'Save changes'
-      expect(@user.reload.attributes.symbolize_keys)
-        .to include first_name: 'Bananas', last_name: 'Smith'
-    end
+    context 'every field is filled in correctly' do
+      let!(:attributes) { attributes_for(:user).except :staff, :last_name }
+      # staff attribute is hidden field, and don't want to change last name
+      before :each do
+        within 'form.edit_user' do
+          fill_in_fields_for User, attributes: attributes
+            .merge(first_name: 'Bananas')
```

Comment:
It's a little awkward to line-break a single method argument. Better to line break at `attributes:`.